File: Sequence Valley/String Matching/Q3. Repeated String Match/Q3_RepeatedStringMatch.java
class Solution {                              // Defines the Solution class
    public int repeatedStringMatch(String a, String b) {  // Method to find minimum repeats of 'a'
        StringBuilder gy = new StringBuilder();            // StringBuilder to build repeated string
        int I = 0;                                        // Variable to count repetitions

        for (I = 1; gy.length() <= b.length(); I++) {     // Repeat until gy is longer than b
            gy.append(a);                                 // Add string 'a' to gy
            if (gy.toString().contains(b))                // Check if b is inside gy
                return I;                                 // Return number of repeats
        }

        if (gy.append(a).toString().contains(b))          // Check once more after extra append
            return I;                                     // Return repeat count

        return -1;                                        // Return -1 if b is never found
    }
}

File: Sequence Valley/String Matching/Q3. Repeated String Match/Q3_RepeatedStringMatch.py
class Solution:                               # Defines the Solution class
    def repeatedStringMatch(self, a, b):      # Method to find minimum repeats of 'a'
        repeat = (len(b) // len(a))            # Estimate minimum repeats needed
        count = 1                              # Start counting repetitions from 1
        while count <= repeat + 2:             # Loop with a small extra buffer
            if b in a * count:                 # Check if b is a substring of repeated a
                return count                   # Return number of repetitions
            else:
                count += 1                     # Increase repeat count
        return -1                              # Return -1 if b is not found
